src/main/apiClient.ts

```diff
@@ -452,14 +510,23 @@ export default class ApiClient {
       .then(() => this.loginWithVerificationCode(params, auth))
       .catch(err => {
         if (err.error) this.eventManager.fireEvent('login_failed', err)
-        throw err
+        return Promise.reject(err)
       })
   }
 
-  signup(params: SignupParams): Promise<void> {
-    const { data, auth, redirectUrl, returnToAfterEmailConfirmation } = params
+  signup(params: SignupParams): Promise<AuthResult | void> {
+    const { data, auth, redirectUrl, returnToAfterEmailConfirmation, saveCredentials } = params
+
```

Comment:
(small suggestion) You can add these variables since they are used twice `l533` and `l546`
```
const { clientId } = this.config
const scope = this.resolveScope(auth)
```

Reply:
Thanks :)